src/l2/eth.rs

```diff
@@ -0,0 +1,28 @@
+use core::mem;
+
+pub const ETH_HDR_LEN: usize = mem::size_of::<EthHdr>();
+
+/// Ethernet header, which is present at the beginning of every Ethernet frame.
+#[repr(C, packed)]
+#[derive(Debug, Copy, Clone)]
+pub struct EthHdr {
+    /// Destination MAC address.
+    pub dest: [u8; 6],
+    /// Source MAC address.
+    pub source: [u8; 6],
+    /// Protocol which is encapsulated in the payload of the frame.
+    pub proto: EthProto,
+}
+
+/// Protocol which is encapsulated in the payload of the Ethernet frame.
+#[repr(u16)]
+#[derive(PartialEq, Eq, Debug, Copy, Clone)]
+pub enum EthProto {
```

Comment:
This should be `EtherType` and is defined by IANA [here](https://www.iana.org/assignments/ieee-802-numbers/ieee-802-numbers.xhtml)